code/scripting/api/libs/engine.cpp

```diff
@@ -58,6 +58,32 @@ ADE_FUNC(addHook,
 		action.hook.override_function.function = override_func;
 	}
 
+	if (action_hook->getDeprecation()) {
+		const auto& deprecation = *action_hook->getDeprecation();
+		bool shownWarn = false;
+		if (deprecation.level_hook == HookDeprecationOptions::DeprecationLevel::LEVEL_ERROR) {
+			LuaError(L, "Hook '%s' is removed since version %s and cannot be used!", action_hook->getHookName().c_str(), gameversion::format_version(deprecation.deprecatedSince).c_str());
+		}
+		else if (deprecation.deprecatedSince <= Targeted_version) {
+			LuaError(L, "Hook '%s' is deprecated since version %s and cannot be used if the mod targets that version or higher!", action_hook->getHookName().c_str(), gameversion::format_version(deprecation.deprecatedSince).c_str());
```

Comment:
I would change "cannot" to "should not" here

Reply:
Hmm, I'm on the fence.
Cause in case it's set to warning mode but the target version is past the deprecation version, it'll error out here. And if it errors out, it kind of can't be used, as the error will stop FSO.

Reply:
You're right, I had mistaken this line for a line that warned instead of errored.  I withdraw my comment.